src/main/java/pl/javastart/task/Memory.java

```diff
@@ -0,0 +1,80 @@
+package pl.javastart.task;
+
+import pl.javastart.task.exception.IllegalParametersException;
+import pl.javastart.task.exception.IllegalTemperatureException;
+import pl.javastart.task.exception.TemperatureOverLimitException;
+
+public class Memory extends Component implements Overclockable {
+
+    private static final int TEMPERATURE_RISE = 15;
+    private static final int OVERCLOCKING_STEP = 100;
+
+    private int size;
+    private int baseFrequency;
+    private double workingTemperature;
+    private double maxSafeTemperature;
+
+    private double overClockedTemperature;
+    private double overClockedFrequency;
```

Comment:
Nie do końca rozumiem to rozróżnienie na dwa taktowania i dwie temperatury. Procesor, czy pamięć ma temperaturę i taktowanie, które zmieniają się przy zwiększaniu, bądź obniżaniu taktowania